receiver/statsdreceiver/transport/transport.go

```diff
@@ -0,0 +1,45 @@
+package transport
+
+import "errors"
+
+// Transport is a set of constants of the transport supported by this receiver.
+type Transport string
+
+var (
+	ErrUnsupportedTransport       = errors.New("unsupported transport")
+	ErrUnsupportedPacketTransport = errors.New("unsupported Packet transport")
+)
+
+const (
+	UDP  Transport = "udp"
+	UDP4 Transport = "udp4"
+	UDP6 Transport = "udp6"
+)
+
+// Create a Transport based on the transport string or return error if it is not supported.
```

Comment:
Nit: It's a good practice to add the function name as the starting comment word (same apply to other comments in this PR) https://go.dev/doc/comment.

Reply:
Thanks!